### ISSUE TITLE
[Bug] Engine configurations not applying gimbal restrictions

### ISSUE THREAD

**Issue by PhineasFreak** [NathanKell/ModularFuelSystem#220]
Real Fuels has the provision in the ModuleEngineConfigs to lock the gimbal capabilities if required. 
This feature does not seem to work, at least in KSP 1.6.1. I created the following simple Module Manager config for the stock SSME to test it:

```
@PART[SSME]
{
    MODULE
    {
        name = ModuleEngineConfigs
        type = ModuleEnginesFX
        configuration = SSME-Gimbal
        origMass = 4.0

        CONFIG
        {
            name = SSME-Gimbal
            minThrust = 0
            maxThrust = 1000
            gimbalRange = 10.5

            PROPELLANT
            {
                name = LiquidFuel
                ratio = 0.9
                DrawGauge = True
            }

            PROPELLANT
            {
                name = Oxidizer
                ratio = 1.1
            }

            atmosphereCurve
            {
                key = 0 315
                key = 1 295
                key = 12 0.001
            }
        }

        CONFIG
        {
            name = SSME-No-Gimbal
            minThrust = 0
            maxThrust = 1000
            gimbalRange = 0.0

            PROPELLANT
            {
                name = LiquidFuel
                ratio = 0.9
                DrawGauge = True
            }

            PROPELLANT
            {
                name = Oxidizer
                ratio = 1.1
            }

            atmosphereCurve
            {
                key = 0 315
                key = 1 295
                key = 12 0.001
            }
        }
    }
}
```

Gimbal is unrestricted for both engine configurations (should be disabled for the second one).

**Mod list:**

* ModuleManager v4.0.2
* RealFuels v12.7.4
* SolverEngines v3.3.0

A similar test was also done under RO (using the built-in engine configs) and the same bug exists (engines like the NK-33 and NK-43 have unrestricted gimbal capabilities even though separate w/gimbal and w/o gimbal).

**Edit:** although not tested no issues have been reported about the KSP 1.3.1 RF releases. Maybe this is something new with KSP 1.4+.

**Comment by Starwaster** [comment]
It's a race condition. ModuleGimbal.gimbalRange is only used once in OnLoad and is used to adjust gimbalRangeXP, gimbalRangeXN, gimbalRangeYP and gimbalRangeYN (gimbal ranges can be independently limited by axis which happens if a given axis is 0 or greater) and all of that is done in OnLoad. Almost certainly before MEC goes to work.

So gimbalRange is probably being set correctly but it's just not being used by then. What I think needs to happen is to set the individual  axes just as ModuleGimbal does under the same conditions that it does as follows:

1. If gimbalRange was present in the MEC then go on to look for the aforementioned individual axes in the MEC.
2. If they are absent then assume that they are -1. If they are present then read their values.
3. Any axis that is -1 well be set to gimbalRange. If the axis was present and 0 or greater then use that value instead.

To sum up, the only way this works is if we set the individual gimbal axes.

**Comment by Starwaster** [comment]
(the individual axis thing was probably added earlier than 1.3.1 back when gimbalRange was used directly to determine the actual range)

**Comment by Starwaster** [comment]
Try using these dlls. It's compiled for 1.7.1 and the test config you provided works properly on my end. The only thing is that individual axes (if configured) won't display in the MEC info display. Also, IF the original engine itself had individual axes configured, those are lost because we have to assume that if gimbalRange is configured in the MEC then we're either retrieving the axis gimbal limits or if absent then they are default values so that gimbalRange will override.

**Comment by Starwaster** [comment]
re-reading that, it's a mess, so to sum up, if gimbalRange is set in the ModularEngineConfigs then the original part's individual axis settings are lost if there were any. However you can also set those in the MEC. (i.e. you can set stuff like gimbalRangeXP BUT gimbalRange also has to be present in the MEC or it won't even bother looking for the individual ones)

**Comment by Starwaster** [comment]
Fixed by commit: 85932921f4272b8664382d2e6868867de7ca0bab
